### PR TITLE
[build-utils] Support directory entries in `Lambda#createZip()`

### PR DESCRIPTION
**Pull request by TooTallNate**
Adds support for empty directory entries in the Lambda `createZip()` function.

### CODE REVIEW

File: packages/build-utils/test/unit.lambda.test.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    expect(linkStat.isSymbolicLink()).toEqual(true);
+    expect(linkDirStat.isSymbolicLink()).toEqual(true);
+    expect(aStat.isFile()).toEqual(true);
+  });
+
+  it('should create zip file with empty directory', async () => {
+    if (process.platform === 'win32') {
+      console.log('Skipping test on windows');
+      return;
+    }
```
Comment (EndangeredMassa) [inline packages/build-utils/test/unit.lambda.test.ts:43]: **suggestion (non-blocking):** remove windows guard

Can this run on Windows?

File: packages/build-utils/test/unit.lambda.test.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          data: 'inside dir b',
+          mode: 33188,
+        }),
+        c: new FileBlob({
+          data: '',
+          mode: 16877,
+        }),
+        'c/a': new FileBlob({
+          data: 'inside dir c',
+          mode: 33188,
```
Comment (EndangeredMassa) [inline packages/build-utils/test/unit.lambda.test.ts]: **suggestion (non-blocking):** label mode values

Can we label these mode values (with comments or named variables) to make it clear what each one means?

**Review by EndangeredMassa — APPROVED** [review]
Can you explain the how this compares to the [the previous PR](https://github.com/vercel/vercel/pull/9103)?

File: packages/build-utils/test/unit.lambda.test.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/build-utils/test/unit.lambda.test.ts:43]: No cause the test relies on the GNU `unzip` command.

**Pull request merged**